webpage/src/App.js

```diff
@@ -1,59 +1,91 @@
 import React, {useState} from 'react'
+import {useSelector, useDispatch} from 'react-redux'
+import fetchRecommendations from './recommendations'
 import './App.css'
 
-function Search() {
+
+const Search = () => {
+  const [query, setQuery] = useState('jess2187')
+  const dispatch = useDispatch()
+  
   return (
-    <form className="Search">
-      <label className="search-box">
-        <input className="search-input" id="search" type="text" placeholder="Username" />
-      </label>
-      {/* <input className="okButton" type="submit" value="Ok" /> */}
-    </form>
+    <form className="Search" 
+        onSubmit={event => {
+          dispatch(fetchRecommendations(query)) 
+          event.preventDefault()
+      }}>
+        <div className="SearchBox">
+          <input
+            className="SearchInput"
+            type="text"
+            value = {query}
+            placeholder="Username"
+            onChange={event => setQuery(event.target.value)}
+          />
+        </div>
+      </form>
   )
 }
 
-function Recommendation(props){
+
+const Recommendations = (props) => {
+  return (
+    props.recs && props.recs.recommendations.map((item, i) => (
+      <Recommendation key ={i} subreddit={item.subreddit} confidence={item.confidence} />
+    ))
+  )
+}
+
+const Recommendation = (props) => {
   return (
     <div className="recommendation">
-      <h3>{"#" + props.score + ":"}</h3>
-      <h3 style={{paddingLeft:"10px"}}>{"r/" + props.subreddit}</h3>
+      <h3>{"r/" + props.subreddit}</h3>
+      <h4>{"Confidence: " + props.confidence}</h4>
     </div>
   )
 }
 
-function App() {
-  const [data, setData] = useState({ hits: [] })
-
-  let recommendations = [
-    {
-      subreddit: "animalcrossing",
-      score: "1"
-    },
-    {
-      subreddit: "nintendoswitch",
-      score: "2"
-    },
-    {
-      subreddit: "thesims",
-      score: "3"
-    }
-  ]
-
-  let recommendationComponents = recommendations.map((item, i) => 
-    <Recommendation key={i} {...item} />
+const Error = (props) => {
```

Comment:
```suggestion
const ErrorPopup = () => (
  <div>
    <h3>Something  went wrong :(</h3>
  </div>
)
```

We can omit return since this is a single expr. Also we'll want to add the actual error message and maybe make this a Modal or Popup.